ocamltest/main.ml

```diff
@@ -221,10 +205,12 @@ let test_file test_filename =
     Printf.eprintf "Wall clock: %s took %.02fs\n%!"
                    test_filename wall_clock_duration
 
-let is_test s =
-  match tsl_block_of_file s with
-  | _ -> true
-  | exception _ -> false
+let is_test filename =
+  let input_channel = open_in filename in
+  let lexbuf = Lexing.from_channel input_channel in
+  Fun.protect ~finally:(fun () -> close_in input_channel) begin fun () ->
```

Comment:
You could use the fancy new stuff: `In_channel.with_open filename @@ fun input_channel ->`.

(Do you want `open_in` or `open_in_bin`?)

Reply:
The first two lines are the same as in the code that does the actual parsing of test scripts. I don't think we want to deviate from that.